OBSDOCS-835: Monitoring 4.16 Release Notes for Monitoring - new features and removed issues (#76755)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

====

[id="ocp-4-16-monitoring-metrics-server-to-access-metrics-api-ga"]
==== Metrics Server component to access the Metrics API general availability (GA)

Metrics Server component is now generally available and automatically installed instead of the deprecated Prometheus Adapter. Metrics Server collects resource metrics and exposes them in the `metrics.k8s.io` Metrics API service for use by other tools and APIs, which frees the core platform Prometheus stack from handling this functionality. For more information, see xref:../observability/monitoring/config-map-reference-for-the-cluster-monitoring-operator.adoc#metricsserverconfig[MetricsServerConfig] in the config map API reference for the Cluster Monitoring Operator.

[id="ocp-4-16-monitoring-new-role-to-allow-read-only-access-to-alertmanager-api"]
==== New monitoring role to allow read-only access to the Alertmanager API

This release introduces new `monitoring-alertmanager-view` role to allow read-only access to the Alertmanager API in the `openshift-monitoring` project.

[id="ocp-4-16-monitoring-vpa-metrics-available-in-kube-state-metrics-agent"]

> ==== VPA metrics are available in Openshift kube-state-metrics agent

Reviewer: ```suggestion
==== VPA metrics are available in OpenShift kube-state-metrics agent
```

> Vertical Pod Autoscaler (VPA) metrics are now available through the `kube-state-metrics` agent. VPA metrics follow a similar exposition format like they did before being deprecated and removed from native support upstream.

Author: **Note to this one**: it needs to be confirmed if we want to include this (generally we do not include Jiras that have `Red Hat Employee` security level.

[id="ocp-4-16-monitoring_{context}"]
=== Monitoring

The in-cluster monitoring stack for this release includes the following new and modified features.

[id="ocp-4-16-monitoring-updates-to-monitoring-stack-components-and-dependencies"]
==== Updates to monitoring stack components and dependencies
This release includes the following version updates for in-cluster monitoring stack components and dependencies:

* kube-state-metrics to 2.12.0
* Metrics Server to 0.7.1
* node-exporter to 1.8.0
* Prometheus to 2.52.0
* Prometheus Operator to 0.73.2
* Thanos to 0.35.0

[id="ocp-4-16-monitoring-changes-to-alerting-rules"]

> ==== Changes to alerting rules

Author: Hello @simonpasquier I remember you mentioned that the team creates a Jira for this. Is the Jira created somewhere? I could not find it and I wanted to update this section.
Thank you!

Reviewer 2: https://issues.redhat.com/browse/MON-3892, I'll try to complete it ASAP.

Author: Oh thank you very much, that will be super helpful :)

Reviewer: summarized the 4.16 alerting rules changes, see from https://issues.redhat.com/browse/MON-3892?focusedId=24948847&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-24948847, we can compare with the result from developers

Author: Thank you @juzhao and @simonpasquier I added the changes to the PR :+1:

[id="ocp-4-16-monitoring_{context}"]
=== Monitoring

> The in-cluster monitoring stack for this release includes the following new and modified features.

Reviewer 3: This sentence can be removed. It would then be consistent with surrounding components.  Although, technically correct to add a lead-in sentence, release notes are that big different. Each note mentions what is happening in this release.

[id="ocp-4-16-monitoring-updates-to-monitoring-stack-components-and-dependencies"]
==== Updates to monitoring stack components and dependencies
This release includes the following version updates for in-cluster monitoring stack components and dependencies:

* kube-state-metrics to 2.12.0
* Metrics Server to 0.7.1
* node-exporter to 1.8.0
* Prometheus to 2.52.0
* Prometheus Operator to 0.73.2
* Thanos to 0.35.0

[id="ocp-4-16-monitoring-changes-to-alerting-rules"]

> ==== Changes to alerting rules

Reviewer 3: ```suggestion
==== New alerting rules
```

[NOTE]
====
Red{nbsp}Hat does not guarantee backward compatibility for recording rules or alerting rules.
====

> * *New*

Reviewer 3: Remove *New* as updates to existing rules is unavailable.

Reviewer 3: Could the note come after the bullet points? The note did not prepare me for the added rules. Maybe afterwards would give customers a moment to orientate themselves with the updates.

Author: This, and the "The in-cluster monitoring stack for this release includes the following new and modified features." semtence is a template we have been using in Monitoring RNs for all the existing doc versions. I will consult with @bburt-rh if we really want to deviate from that, thank you!

Reviewer 4: @eromanova97 @dfitzmau - My take is that it is fine as-is to stay consistent with the way the monitoring RNs have been structured since at least the 4.9 release. The same applies to the previous comment.

** Added the `ClusterMonitoringOperatorDeprecatedConfig` alert to monitor when the Cluster Monitoring Operator configuration uses a deprecated field.
** Added the `PrometheusOperatorStatusUpdateErrors` alert to monitor when the Prometheus Operator fails to update object status.

[id="ocp-4-16-monitoring-metrics-server-to-access-metrics-api-ga"]

> ==== Metrics Server component to access the Metrics API general availability (GA)

Reviewer 3: ```suggestion
==== Metrics Server component to access the Metrics API
```

Author: I see in release notes we normally specify in the title that the feature is not generally availible, either like this:
* https://docs.openshift.com/container-platform/4.15/release_notes/ocp-4-15-release-notes.html#ocp-4-15-storage-retro-sc-assignment

Or the way I did it:
* https://docs.openshift.com/container-platform/4.15/release_notes/ocp-4-15-release-notes.html#ocp-4-15-networking-ovn-kubernetes-ipsec-support-for-external-traffic

So I believe it should stay as it it.

Metrics Server component is now generally available and automatically installed instead of the deprecated Prometheus Adapter. Metrics Server collects resource metrics and exposes them in the `metrics.k8s.io` Metrics API service for use by other tools and APIs, which frees the core platform Prometheus stack from handling this functionality. For more information, see xref:../observability/monitoring/config-map-reference-for-the-cluster-monitoring-operator.adoc#metricsserverconfig[MetricsServerConfig] in the config map API reference for the Cluster Monitoring Operator.

[id="ocp-4-16-monitoring-new-role-to-allow-read-only-access-to-alertmanager-api"]

> ==== New monitoring role to allow read-only access to the Alertmanager API

Reviewer 3: ```suggestion
==== New monitoring role to allow `read-only` access to the Alertmanager API
```

Author: Same as the previous comment

This release introduces a new `monitoring-alertmanager-view` role to allow read-only access to the Alertmanager API in the `openshift-monitoring` project.

[id="ocp-4-16-monitoring-vpa-metrics-available-in-kube-state-metrics-agent"]

> ==== VPA metrics are available in OpenShift kube-state-metrics agent

Reviewer 3: ```suggestion
==== VPA metrics are available in the `kube-state-metrics` agent
```

Author: Based on OpenShift guidelines, we are not supposed to use backticks in headings, that should imply for RN titles, if I am not mistaken. I see it like this in the existing examples:
* "Introduction of HAProxy no strict-limits variable" ([link](https://docs.openshift.com/container-platform/4.15/release_notes/ocp-4-15-release-notes.html#ocp-4-15-notable-technical-changes))
Instead of "Introduction of HAProxy no `strict-limits` variable"
Or "Must-gather storage limit default" instead of "`must-gather` storage limit default"

So I am retaining the existing version (I will fix the article though).

==== New monitoring role to allow read-only access to the Alertmanager API

This release introduces a new `monitoring-alertmanager-view` role to allow read-only access to the Alertmanager API in the `openshift-monitoring` project.

[id="ocp-4-16-monitoring-vpa-metrics-available-in-kube-state-metrics-agent"]
==== VPA metrics are available in the kube-state-metrics agent

Vertical Pod Autoscaler (VPA) metrics are now available through the `kube-state-metrics` agent. VPA metrics follow a similar exposition format like they did before being deprecated and removed from native support upstream.

[id="ocp-4-16-monitoring-change-in-proxy-service-for-monitoring-components"]
==== Change in proxy service for monitoring components
With this release, the proxy service in front of Prometheus, Alertmanager, and Thanos Ruler has been updated from OAuth to `kube-rbac-proxy`. This change might affect service accounts and users accessing these API endpoints without the appropriate roles and cluster roles.

[id="ocp-4-16-monitoring-change-in-how-prometheus-handles-duplicate-samples"]
==== Change in how Prometheus handles duplicate samples

> With this release, when Prometheus scrapes a target, duplicate samples are no longer silently ignored, even if they have the same value. The first sample is accepted and the `prometheus_target_scrapes_sample_duplicate_timestamp_total` counter is incremented, which might trigger the `PrometheusDuplicateTimestamps` alert.

Reviewer: I don't think we need to change, we can keep as it was.
reason:
4.15 and 4.16, PrometheusDuplicateTimestamps alert is the same definition. I think the purpose to mention the change for PrometheusDuplicateTimestamps alert is it will add runbook_url after https://github.com/openshift/cluster-monitoring-operator/pull/2365 back ported to 4.16. @machine424 what's your suggestion
```
$ oc get clusterversion
NAME      VERSION                              AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.15.0-0.nightly-2024-06-19-045027   True        False         10h     Cluster version is 4.15.0-0.nightly-2024-06-19-045027

$ oc -n openshift-monitoring get prometheusrules prometheus-k8s-prometheus-rules -ojsonpath='{.spec.groups[].rules[?(@.alert=="PrometheusDuplicateTimestamps")]}' | jq
{
  "alert": "PrometheusDuplicateTimestamps",
  "annotations": {
    "description": "Prometheus {{$labels.namespace}}/{{$labels.pod}} is dropping {{ printf \"%.4g\" $value  }} samples/s with different values but duplicated timestamp.",
    "summary": "Prometheus is dropping samples with duplicate timestamps."
  },
  "expr": "rate(prometheus_target_scrapes_sample_duplicate_timestamp_total{job=~\"prometheus-k8s|prometheus-user-workload\"}[5m]) > 0\n",
  "for": "1h",
  "labels": {
    "severity": "warning"
  }
}
```

> [id="ocp-4-16-monitoring-change-in-how-prometheus-handles-duplicate-samples"]
> ==== Change in how Prometheus handles duplicate samples
> With this release, when Prometheus scrapes a target, duplicate samples are no longer silently ignored, even if they have the same value. The first sample is accepted and the `prometheus_target_scrapes_sample_duplicate_timestamp_total` counter is incremented, which triggers the `PrometheusDuplicateTimestamps` alert. The attached runbook should help the user resolve the issue.

Reviewer 2: "attached runbook" => do we mean that users should open the runbook URL? I assume that we can't link it directly here?
```suggestion
[id="ocp-4-16-monitoring-change-in-how-prometheus-handles-duplicate-samples"]
==== Change in how Prometheus handles duplicate samples
With this release, when Prometheus scrapes a target, duplicate samples are no longer silently ignored, even if they have the same value. The first sample is accepted and the `prometheus_target_scrapes_sample_duplicate_timestamp_total` counter is incremented, which may trigger the `PrometheusDuplicateTimestamps` alert. The attached runbook should help the user resolve the issue.
```

Author: Correct, I changed the wording of the last sentence to be more clear, thank you!

> With this release, when Prometheus scrapes a target, duplicate samples are no longer silently ignored, even if they have the same value. The first sample is accepted and the `prometheus_target_scrapes_sample_duplicate_timestamp_total` counter is incremented, which might trigger the `PrometheusDuplicateTimestamps` alert. The runbook linked to that alert should help the user resolve the issue.

Reviewer 3: "Should" is one of those tricky words. How about:

> A user can refer to the runbook that is linked to an alert for help on how to resolve the issue.

![Screenshot from 2024-06-20 09-05-24](https://github.com/openshift/openshift-docs/assets/57954076/5e531cf2-5953-448d-abcf-58247762c85b)

> With this release, when Prometheus scrapes a target, duplicate samples are no longer silently ignored, even if they have the same value. The first sample is accepted and the `prometheus_target_scrapes_sample_duplicate_timestamp_total` counter is incremented, which might trigger the `PrometheusDuplicateTimestamps` alert. A user can refer to the runbook that is linked to the alert for help on how to resolve the issue.

Reviewer: `A user can refer to the runbook that is linked to the alert for help on how to resolve the issue`?
do you mean the runbook_url is added to `PrometheusDuplicateTimestamps` alert? but did not find the runbook_url 
```
$ oc -n openshift-monitoring get prometheusrules prometheus-k8s-prometheus-rules -ojsonpath='{.spec.groups[].rules[?(@.alert=="PrometheusDuplicateTimestamps")]}' | jq
{
  "alert": "PrometheusDuplicateTimestamps",
  "annotations": {
    "description": "Prometheus {{$labels.namespace}}/{{$labels.pod}} is dropping {{ printf \"%.4g\" $value  }} samples/s with different values but duplicated timestamp.",
    "summary": "Prometheus is dropping samples with duplicate timestamps."
  },
  "expr": "rate(prometheus_target_scrapes_sample_duplicate_timestamp_total{job=~\"prometheus-k8s|prometheus-user-workload\"}[5m]) > 0\n",
  "for": "1h",
  "labels": {
    "severity": "warning"
  }
}
```

Reviewer: I see maybe you meant https://github.com/openshift/cluster-monitoring-operator/pull/2365 which would be merged soon

Author: Ohh I see, @machine424 Should we remove the last sentence as the above PR was still not merged?

Reviewer 5: As I mentioned [here](https://issues.redhat.com/browse/OCPBUGS-35397?focusedId=24963680&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-24963680), I was willing to make sure the code and doc are synced. If you want to add that sentence once the fix is merged, we can remove it.
Maybe we don't even need it (users will spontaneously look for the runbook)

Author: Thank you for clarification! I missed that last comment in the Jira :+1:  I think we can remove the last sentence as you suggest :+1:

Reviewer 5: no, nothing else.

Author: Okay so based on the comments here and in Slack, I am not sure what we want to do here, do we want to add the sentence about the runbook back? (I thought the main thing here was to highlight the new improved behavior)

Reviewer 5: no need to add the runbook phrase back.

Reviewer: got it, thanks!
https://github.com/prometheus/prometheus/releases/tag/v2.52.0
```
[ENHANCEMENT] Scrape: Increment prometheus_target_scrapes_sample_duplicate_timestamp_total metric on duplicated series during one scrape. https://github.com/prometheus/prometheus/pull/12933
```
OCP 4.16 uses prometheus v2.52.0, included the enhancement
https://github.com/openshift/prometheus/blob/release-4.16/CHANGELOG.md
```
[ENHANCEMENT] Scrape: Increment prometheus_target_scrapes_sample_duplicate_timestamp_total metric on duplicated series during one scrape. #12933
```
